controllers/clustertopology_controller_compute.go

```diff
@@ -308,6 +332,101 @@ func computeCluster(current *clusterTopologyState, infrastructureCluster, contro
+		cluster:               current.cluster,
+		namePrefix:            fmt.Sprintf("%s-%s-infra-", current.cluster.Name, machineDeploymentTopology.Name),
+		currentObjectRef:      machineDeploymentClass.Template.Infrastructure.Ref,
+		labels:                mergeMap(machineDeploymentTopology.Metadata.Labels, machineDeploymentClass.Template.Metadata.Labels),
+		annotations:           mergeMap(machineDeploymentTopology.Metadata.Annotations, machineDeploymentClass.Template.Metadata.Annotations),
+	})
+
+	machineDeployment := computeMachineDeployment(current, currentMachineDeployment, machineDeploymentTopology, machineDeploymentClass)
+	machineDeployment.Spec.Template.Spec.Bootstrap.ConfigRef = objToRef(state.bootstrapTemplate)
+	machineDeployment.Spec.Template.Spec.InfrastructureRef = *objToRef(state.infrastructureMachineTemplate)
+	state.object = machineDeployment
+
+	return state, nil
+}
+
+// computeMachineDeployment computes a MachineDeployment object for a topology defined in the Cluster object
+// The generated machineDeployment object is calculated using the values from the machineDeploymentTopology and
+// the machineDeployment class.
+func computeMachineDeployment(cluster *clusterTopologyState, current *clusterv1.MachineDeployment, topology clusterv1.MachineDeploymentTopology, class *clusterv1.MachineDeploymentClass) *clusterv1.MachineDeployment {
+	gv := clusterv1.GroupVersion
+	machineDeployment := &clusterv1.MachineDeployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       gv.WithKind("MachineDeployment").Kind,
+			APIVersion: gv.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      names.SimpleNameGenerator.GenerateName(fmt.Sprintf("%s-%s-", cluster.cluster.Name, topology.Name)),
+			Namespace: cluster.cluster.Namespace,
+		},
+		Spec: clusterv1.MachineDeploymentSpec{
+			ClusterName: cluster.cluster.Name,
+			Template: clusterv1.MachineTemplateSpec{
+				Spec: clusterv1.MachineSpec{
+					ClusterName: cluster.cluster.Name,
+					Version:     pointer.String(cluster.cluster.Spec.Topology.Version),
+				},
+			},
+		},
+	}
+
+	// if an existing machine deployment is present, override the machined deployment
+	// object with the same name.
+	if current != nil {
+		machineDeployment.SetName(current.Name)
+	}
+
+	labels := mergeMap(topology.Metadata.Labels, class.Template.Metadata.Labels)
+	labels[clusterv1.ClusterLabelName] = cluster.cluster.Name
+	labels[clusterv1.ClusterTopologyLabelName] = ""
+	labels[clusterv1.MachineDeploymentTopologyLabelName] = topology.Name
+	machineDeployment.SetLabels(labels)
+
+	machineDeployment.Annotations = mergeMap(topology.Metadata.Annotations, class.Template.Metadata.Annotations)
+
+	if topology.Replicas != nil {
+		replicas := pointer.Int32(int32(*topology.Replicas))
+		machineDeployment.Spec.Replicas = replicas
+	}
+	return machineDeployment
+}
+
+func getMachineDeploymentClass(class *clusterTopologyClass, name string) *clusterv1.MachineDeploymentClass {
```

Comment:
What about 
```suggestion
func (class *clusterTopologyClass) getMachineDeploymentByName(name string) *clusterv1.MachineDeploymentClass {
```
So it is consistent with what implemented for the topology state?

Reply:
I am seeing where you are going with the naming scheme, but for this instance, the name of the method should reflect that we are looking for the MachineDeploymentClass (& not MachineDeployment).